typegen/package.json

```diff
@@ -0,0 +1,15 @@
+{
+  "name": "typegen",
+  "version": "1.0.0",
+  "description": "Generates TypeScript definition file",
+  "main": "generate-types.js",
+  "scripts": {
+    "generate-types": "node generate-types.js",
```

Comment:
maybe make this babel-node instead of node. I just tried running this locally (i'm on node v8) and it failed

Reply:
What error did you get? Should work well in node v8.10 and above.
This tool loads built version of react-vis from the **dist** folder:
`const reactVis = require('../dist');` 
Maybe problem is here. Have you tried to run build from project root? It should run generate-types as final step of build

Reply:
Here's the error i got, I ran the generate-type script from package.json

```sh
 mcnutt$ npm run generate-types

> react-vis@1.11.2 generate-types /Users/mcnutt/react-vis
> cd typegen && npm install && npm run generate-types && npm run generate-modules

npm WARN typegen@1.0.0 No repository field.

audited 1 package in 2.894s
found 0 vulnerabilities


> typegen@1.0.0 generate-types /Users/mcnutt/react-vis/typegen
> node generate-types.js

/Users/mcnutt/react-vis/typegen/generate-types.js:11
  ...reactVis.AbstractSeries.propTypes,
  ^^^

SyntaxError: Unexpected token ...
    at createScript (vm.js:74:10)
    at Object.runInThisContext (vm.js:116:10)
    at Module._compile (module.js:533:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! typegen@1.0.0 generate-types: `node generate-types.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the typegen@1.0.0 generate-types script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mcnutt/.npm/_logs/2018-10-18T03_55_21_272Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! react-vis@1.11.2 generate-types: `cd typegen && npm install && npm run generate-types && npm run generate-modules`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the react-vis@1.11.2 generate-types script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mcnutt/.npm/_logs/2018-10-18T03_55_21_362Z-debug.log
```

Reply:
I have refactored code in order to be compatible with node v6.14+

Reply:
That's really good, but I think it would be better to keep out scripting strategy consistent across the library. I think a better strategy would be to remove the package.json from /typegen and merge it with the main package.json (using prop-types-to-dts as a dev dep) and placing the generate-types command in scripts section of package. I tried this approach out locally and it appears to work well.